### DISCUSSION TITLE
Future of utilities

### DISCUSSION THREAD

**Discussion by leighkendell** [vanilla-extract-css/vanilla-extract#29] [Q&A] [answered]
Hi! Loving what vanilla-extract is capable of so far. I've started to adopt it in a styled-components project making use of the CSS custom property features so far.

Are there any plans to add additional helper functions, for example returning the "key" of a custom property i.e. "--my-var12321" instead of the version wrapped in "var(...)". I'll probably regex this for now as I'd like to use the value to override on a per component basis using the style attribute. I've had a look at the dynamic API but this seems somewhat redundant inside of a React context, would love to hear how you're currently using it with react in general too :) 

Cheers

**Comment by markdalgleish** [comment, marked as answer]
We could provide a `getVarName` function as part of the `dynamic` package, which is currently a private util.

We could also potentially add a higher level function that sets a single variable as an inline style object, as we currently only support setting entire themes as inline styles.

**Reply by leighkendell** [reply]
That would be useful! Being able to override individual variables and output the inline style sounds just like what I need :)
